ping/datadog_checks/ping/ping.py

```diff
@@ -26,25 +26,30 @@ def _load_conf(self, instance):
 
     def _exec_ping(self, timeout, target_host):
         if platform.system() == "Windows":  # pragma: nocover
+            precmd = ["cmd", "/c", "chcp 437 &"]  # Set code page to English for non-US Windows
```

Comment:
```suggestion
    def _exec_ping(self, timeout, target_host):
        precmd = []
        if platform.system() == "Windows":  # pragma: nocover
            precmd = ["cmd", "/c", "chcp 437 &"]  # Set code page to English for non-US Windows
```

Instead of setting precmd to `[]` at every logic branch, perhaps you can just set it at the beginning of the function and just update for the Windows branch. This reduces repetitive code.